tools/cachegrand-benchmark-generator/src/support.c

```diff
@@ -13,42 +13,48 @@
 #include "support.h"
 
 char* support_read_file(
-        char *filename) {
+        const char *filename) {
     char *buffer = NULL;
-    int string_size, read_size;
+    size_t string_size, read_size = 0;
     FILE *handler = fopen(filename, "r");
 
     if (handler) {
         fseek(handler, 0, SEEK_END);
         string_size = ftell(handler);
         rewind(handler);
 
-        buffer = (char*) malloc(sizeof(char) * (string_size + 1) );
-        read_size = fread(buffer, sizeof(char), string_size, handler);
+        buffer = (char*) malloc(sizeof(char) * (string_size + 1));
+        do {
+            size_t read_size_temp = fread(buffer, sizeof(char), string_size, handler);
+            if (read_size_temp <= 0) {
+                free(buffer);
+                buffer = NULL;
+                perror("failed to read file");
```

Comment:
better you add a return NULL, an exit or something otherwise this loop will go ahead for forever

the return value depends on the caller as it needs to be aware that there is a failure, if you don't have such logic in place potentially it's fine to do a straight exit 1 and simply abort everything